### PR TITLE
feat(global): multi proxy provider

### PR DESCRIPTION
**Pull request by rimiti**
## PURPOSE

This PR aims to offer the possibility to use Luminati or ProxyRack proxy provider.

### CODE REVIEW

File: src/classes/providers/proxyrack/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    return this.sendRequest(axiosRequestConfig);
+  }
+
+  /**
+   * @description Create and set proxy agents.
+   * @returns {Proxyrack}
+   */
+  setIp(): Proxyrack {
+    const { httpAgent, httpsAgent } = this.createProxyAgents();
```
Comment (Theo-Le-Donne) [inline src/classes/providers/proxyrack/index.ts:45]: typing missing

File: src/classes/providers/proxyrack/index.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { HttpProxyAgent } from 'http-proxy-agent';
+import { ICreateProxyConfig } from '../base/types';
+import { replacer } from '../../../utils/replacer';
+import { Base } from '../base';
+import { IProxyrackConfig } from './types';
+
+/**
+ * @description Proxyrack proxy provider.
+ */
+export class Proxyrack extends Base {
```
Comment (Theo-Le-Donne) [inline src/classes/providers/proxyrack/index.ts:12]: 👍

File: src/classes/providers/luminati/index.ts [hunk 3]
```diff
@@ -66,9 +57,9 @@ export class Luminator {
 
     // Creates an agent with specific countries and a specific sessionId
     if (params.countries && params.sessionId) {
-      Luminator.checkIfCountriesArrayIsntEmpty(params.countries);
+      Luminati.checkIfCountriesArrayIsntEmpty(params.countries);
 
-      const { httpAgent, httpsAgent }: ICreateProxyAgents = this.createProxyAgents({
+      const { httpsAgent, httpAgent } = this.createProxyAgents({
```
Comment (Theo-Le-Donne) [inline src/classes/providers/luminati/index.ts:62]: typing missing

File: package.json [hunk 4]
```diff
@@ -10,20 +10,20 @@
     "@rimiti/stimmy": "1.8.0",
     "axios": "0.21.0",
     "http-proxy-agent": "4.0.1",
-    "https-proxy-agent": "5.0.0"
+    "https-proxy-agent": "2.2.4"
```
Comment (Theo-Le-Donne) [inline package.json:13]: 👍

**Review by Theo-Le-Donne — APPROVED** [review]
minor remarks, LGTM !

**Review by Simobain — APPROVED** [review]
LGTM !

**Pull request merged**